src/ResourceManager/Websites/Commands.Websites/Cmdlets/BackupRestore/RestoreAzureDeletedWebApp.cs

```diff
@@ -0,0 +1,105 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.WebApps.Cmdlets.WebApps;
+using Microsoft.Azure.Commands.WebApps.Models;
+using Microsoft.Azure.Management.WebSites.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.BackupRestore
+{
+    /// <summary>
+    /// Restores a deleted Azure Web App's contents and settings to an existing web app
+    /// </summary>
+    [Cmdlet("Restore", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeletedWebApp"), OutputType(typeof(void))]
+    public class RestoreAzureDeletedWebApp : WebAppBaseClientCmdLet
+    {
+        // Note, there can be multiple deleted web apps with the same name and resource group. Internally each deleted app has a unique ID.
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "The deleted Azure Web App.", ValueFromPipeline = true)]
+        public AzureDeletedWebApp InputObject;
```

Comment:
Looks like you only did the inputobject parameter set - you need the interactive one as well (as discussed in the design review)

```
Restore-AzureRmDeletedWebApp -ResourceGroupName <string> -Name <string> [-TargetResourceGroupName <string>] -TargetName <string>
```

Reply:
Sorry, I overlooked that. The problem is that there could be a web app with the same name and resource group that was deleted and re-created multiple times. This is actually a relatively common case. In the backend we've given each deleted app a unique ID to avoid ambiguity.

We could default to restoring the most recent deleted app with the provided name and resource group. Do you think that would be good, or could it be too confusing?

Reply:
Hmm I do see your point.  I think it would be good to be restore the most recently deleted app, but add a warning message if there were multiple stating that we chose to restore the most recent app.  Could you also add a parameter set to accept that unique id? Or do you think that the users would not ever want to use that functionality? Apologies if we've previously discussed this.

Reply:
I'm working on adding the option to undelete by deleted site name with your suggestions.

I don't think many people would use a deleted site ID parameter. The ID is a generated integer that's only discoverable by using the Get-AzureRmDeletedWebApp cmdlet. It's easiest to just use the deleted site object directly instead of pulling the ID out of it. The ID is also something we'd prefer to keep somewhat hidden from customers because it's not much use to them.